fiftyone/utils/eval/coco.py

```diff
@@ -89,39 +111,295 @@ def evaluate_image(
+            matches. Either label can be ``None`` to indicate an unmatched
+            object
+        precision: an array of precision values of shape
+            ``num_iou_threshs x num_classes x num_recall``
+        recall: an array of recall values
+        iou_threshs: the list of IoU thresholds
+        classes: the list of possible classes
+        missing ("none"): a missing label string. Any unmatched objects are
+            given this label for evaluation purposes
+    """
+
+    def __init__(
+        self, matches, precision, recall, iou_threshs, classes, missing="none"
+    ):
+        super().__init__(matches, classes=classes, missing=missing)
+        self.precision = precision
+        self.recall = recall
+        self.iou_threshs = iou_threshs
+        self._classwise_AP = np.mean(precision, axis=(0, 2))
+
+    def plot_pr_curves(self, classes=None, ax=None, block=False, **kwargs):
+        """Plots precision-recall (PR) curves for the detection results.
+
+        Args:
+            classes (None): a list of classes to generate curves for. By
+                default, top 3 AP classes will be plotted
+            ax (None): an optional matplotlib axis to plot in
+            block (False): whether to block execution when the plot is
+                displayed via ``matplotlib.pyplot.show(block=block)``
+            **kwargs: optional keyword arguments for
+                ``sklearn.metrics.PrecisionRecallDisplay.plot(**kwargs)``
+
+        Returns:
+            the matplotlib axis containing the plots
+        """
+        if not classes:
+            inds = np.argsort(self._classwise_AP)[::-1][:3]
+            classes = [self.classes[i] for i in inds]
+
+        for c in classes:
+            class_ind = self.classes.index(c)
+            precision = np.mean(self.precision[:, class_ind], axis=0)
+            avg_precision = np.mean(precision)
+            display = skm.PrecisionRecallDisplay(
+                precision=precision, recall=self.recall
+            )
+            label = "AP = %.2f, class = %s" % (avg_precision, c)
+            display.plot(ax=ax, label=label, **kwargs)
+            ax = display.ax_
+
+        plt.show(block=block)
+
+        return ax
+
+    def mAP(self, classes=None):
+        """Computes COCO-style mean average precision (mAP) for the specified
+        classes.
+
+        See `this page <https://github.com/cocodataset/cocoapi/blob/master/PythonAPI/pycocotools/cocoeval.py>`_
+        for more details about COCO-style mAP.
+
+        Args:
+            classes (None): a list of classes for which to compute mAP
+
+        Returns:
+            the mAP float
+        """
+        if classes is not None:
+            class_inds = [self.classes.index(c) for c in classes]
+            classwise_AP = self._classwise_AP[class_inds]
+        else:
+            classwise_AP = self._classwise_AP
+
+        existing_classes = classwise_AP > -1
+        if not any(existing_classes):
```

Comment:
you wrote `len(existing_classes) == 0` here, but I think you meant `not any(existing_classes)` right?

Reply:
Ah yup!